native/core/src/execution/planner.rs

```diff
@@ -956,6 +956,12 @@ impl PhysicalPlanner {
                     self.create_expr(filter.predicate.as_ref().unwrap(), child.schema())?;
 
                 let filter: Arc<dyn ExecutionPlan> = if filter.use_datafusion_filter {
+                    assert_eq!(
+                        scans.len(),
+                        0,
+                        "Must not use ScanExec as it may requires a copy"
+                    );
+
```

Comment:
My understanding is that we only reuse buffers when using `native_comet` to read the Parquet files. If `native_iceberg_compat` is used then we will still have a `ScanExec` reading from JVM but it is safe to use DataFusion's FilterExec in this case.

@parthchandra would you agree with this?